javalib/src/main/scala/java/io/DataOutputStream.scala

```diff
@@ -1,8 +1,14 @@
 package java.io
 
+import java.nio.charset.StandardCharsets
+
 class DataOutputStream(out: OutputStream)
     extends FilterOutputStream(out)
     with DataOutput {
+  // Capacity is a guess: Balance memory use & execution speed.
+  // Allow small to moderate sized Strings to be written as Chars in one shot.
+  private val byteArrayCapacity = 1024 * 2
+  private val byteArray         = Array.ofDim[Byte](byteArrayCapacity)
```

Comment:
Could we call this `buffer` instead? Also, use `new Array[Byte]`:
```suggestion
  private val buffer         = new Array[Byte](byteArrayCapacity)
```

Reply:
Done. Is the suggested change because `Array.ofDim`  is Scala, not Java?  The former
seems to creep in quite easily.

Reply:
No, just because `Array.ofDim` is a completely useless bag of overhead, both at compile-time and at run-time. Even in a regular Scala codebase, I would recommend against using it.